postgraphiql/src/components/PostGraphiQL.js

```diff
@@ -350,9 +352,9 @@ class PostGraphiQL extends React.PureComponent {
             });
           };
 
-          const subscription = this.subscriptionsClient.request(graphQLParams).subscribe(observer);
+          const unsubscribe = this.subscriptionsClient.subscribe(graphQLParams, observer);
```

Comment:
I'd preserve the existing API here; even if `unsubscribe` is the only method this exposes currently, there may be the need to add additional helpers in future, and adding them as properties of an unsubscribe function is kinda ugly.

Reply:
What API are you talking about exactly? The `subscribe` method intentionally returns **just** a cleanup function because it is intended to be a "subscriber" of an [Observable](https://github.com/tc39/proposal-observable#observable).

```ts
const observable = Observable.create(
  function subscriberFunction(observer: SubscriptionObserver) {
    // Subscriber func returns a cleanup function which will cancel the event stream/dispose of the subscription
    return subscriptionsClient.subscribe(operation, observer);
  }
);
```

Reply:
I thought Observable's returned a `Subscription` object that has both an `unsubscribe()` method and other details (e.g. the `closed` getter):

```
interface Subscription {

    // Cancels the subscription
    unsubscribe() : void;

    // A boolean value indicating whether the subscription is closed
    get closed() : Boolean;
}
```

You seem to be returning a function directly? I'm just guessing.

Reply:
Yes, the subscriber returns a cleanup function directly. Its the standard; RxJS, Relay, Apollo, CoreJS and even the [Observable proposal](https://github.com/tc39/proposal-observable) have the following signature:
```ts
function listen(element, eventName) {
    return new Observable(observer => {
        // Create an event handler which sends data to the sink
        let handler = event => observer.next(event);

        // Attach the event handler
        element.addEventListener(eventName, handler, true);

        // Return a cleanup function which will cancel the event stream
        return () => {
            // Detach the event handler from the element
            element.removeEventListener(eventName, handler, true);
        };
    });
}
```

*But*, what you might be confused with is expecting the `subscribe` method **to return an Observable** (which has the `unsubscribe` method). This is **not the case**, it is actually the subscriber (the implementing/constructor part of an Observable). You essentially make an Observable **using** the `subscribe`. This way the lib is completely agnostic to any observable pattern / streaming implementation the user (or a GQL client) might want to use.

Here is an example on how to make an Observable:
```ts
import { Observable } from 'relay-runtime';
// or
import { Observable } from '@apollo/client';
// or
import { Observable } from 'rxjs';
// or
import Observable from 'zen-observable';

function request(operation) {
  return Observable.create((observer) => {
    // subscriber implementation
    return subscriptionsClient.subscribe(operation, observer);
  });
}

const subscription = request({ ... } as Operation);
subscription.unsubscribe(); // ✅
```